Amplify/Categories/DataStore/Model/Temporal/Time.swift

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+extension Temporal {
+
+    /// `Time` is an immutable `TemporalSpec` object that represents a time, often viewed
+    /// as hour-minute-second (`HH:mm:ss`). Time can be represented to nanosecond precision.
+    /// For example, the value "13:45.30.1234". It can also hold a reference to a TimeZone.
+    ///
+    /// As all Date scalars, `Time` relies on the ISO8601 calendar and fixed format.
+    public struct Time: TemporalSpec, TimeUnitOperable {
+
+        public static var iso8601DateComponents: Set<Calendar.Component> {
+            [.hour, .minute, .second, .nanosecond, .timeZone]
+        }
+
+        public static func now() -> Time {
+            return Time(Foundation.Date())
+        }
+
+        public let date: Foundation.Date
+
+        public init(_ date: Foundation.Date) {
+            // sets the date to a fixed instant so time-only operations are safe
+            let calendar = Time.iso8601Calendar
+            var components = calendar.dateComponents(
+                [.year, .month, .day, .hour, .minute, .second, .nanosecond, .timeZone],
+                from: date
+            )
+            components.year = 2_000
```

Comment:
For this case specifically no because that's what `Foundation.Date` does as well if you parse a time-only string (sets the date to `2000-01-01`). But for the `Temporal.Date` one it might not be the case.